README.md

```diff
@@ -15,40 +15,56 @@ Run `make test` to run all the tests.
 
 ## Redis Source
 
-The redis source has 2 modes
+The redis connector watches for new data being added in a *single* redis key supplied in `redis.key`. 
+Currently, the connector supports two type of Redis Data structures: `pubsub` & `stream`.
+To decide which type of DS the redis key holds, the `mode` setting is used. 
+The connector by default starts in `pubsub` mode and subscribes to the channel provided in `redis.key` settings using `SUBSCRIBE <redis.key>`
+To start stream iterator pass `stream` as mode value.
 
-### MODE_PUBSUB
+**Q. Why can't we use `TYPE <key>` command to decide which iterator to start?**
+A. There are 2 reasons for that:
+- The pubsub channel names can conflict with existing keys of other DS types. 
+For ex: there can be an existing key named `mystream` which holds stream DS and still the same name can be used as channel name
+- `TYPE` command on channel name returns `none`
 
-In this mode the source subscribes to the channel provided in key and starts listening for new messages published on the channel.
+
+### Mode: pubsub
+
+In this mode the source subscribes to the channel provided in `redis.key` setting, during configuration, and starts listening for new messages published on the channel.
 The listener will stop only when the pipeline is paused or any error is encountered.
+**NOTE:** The connector doesn't support pattern based channel subscription, it only subscribes to a single channel using `SUBSCRIBE <key>`
```

Comment:
What's the reason for this limitation? May I ask you to add it to a list of known limitations (if not already)?